core/src/test/scala/org/apache/spark/sql/delta/LogStoreSuite.scala

```diff
@@ -301,6 +301,30 @@ trait HDFSLogStoreSuiteBase extends LogStoreSuiteBase {
     }
   }
 
+  test("Read after write consistency with msync") {
+     withTempDir { tempDir =>
+      val tsFSLocation = s"ts://${tempDir.getCanonicalFile}"
+      // Use the file scheme so that it uses a different FileSystem cached object
+      withSQLConf(
+        ("fs.ts.impl", classOf[TimestampLocalFileSystem].getCanonicalName),
+        ("fs.AbstractFileSystem.ts.impl",
+          classOf[TimestampAbstractFileSystem].getCanonicalName)) {
+        val store = createLogStore(spark)
+        val path = new Path(tsFSLocation, "1.json")
+
+        // Initialize the TimestampLocalFileSystem object which will be reused later due to the
+        // FileSystem cache
+        assert(store.listFrom(path, sessionHadoopConf).length == 0)
+        // The LocalFileSystem only tracks modified time at second granularity, so we need to
+        // wait a second to make sure the behavior works
+        Thread.sleep(1000)
```

Comment:
why do we need `sleep` here?

Reply:
Basically (at least on my machine) the RawLocalFileSystem's FileStatus.getModificationTime is only returning second-granular times. Without it I ran into the issue of:

- FileSystem gets initialized and sets `latestTimestamp` to 53217 milliseconds
- `store.write` creates the file at 53321 milliseconds
- `listStatus` has a `getModificationTime` of that file as 53000
- 53000 < 53217 so it's not considered "stale" even though it was written after 53217

So the sleep makes sure the modification time at least gets to 54000. Obviously those numbers are made up. Tried to convey that in the comment, let me know if you can think of a better description

Reply:
Added a little more to the comment